gui/gui/src/TRootDialog.cxx

```diff
@@ -142,11 +144,23 @@ const char *TRootDialog::GetParameters()
       }
 
       if (params.Length()) params += ",";
-      if (type && data) {
-         if (!strncmp(type, "char*", 5))
+      TDataType  *datatype = gROOT->GetType(type);
+      if (datatype && type && data) {
+         param = TString::Format("(%s)%s", type, data);
+      } else if (type && data) {
+         if (!strncmp(type, "char*", 5) || !strncmp(type, "const char*", 11))
             param = TString::Format("\"%s\"", data);
-         else
-            param = TString::Format("(%s)%s", type, data);
+         else {
+            ULong_t address = (Long_t)gROOT->FindObject(data);
+            if (address == 0) {
+               TGlobal* gb = (TGlobal*)gROOT->GetListOfGlobals(kTRUE)->FindObject(data);
+               if (gb)
+                  address = (ULong_t)gb->GetAddress();
+               if (address == -1)
+                  address = 0;
+            }
+            param = TString::Format("(%s)0x%lx", type, address);
```

Comment:
What is the user types in the box:
```
h1arr[1]
```
or some other expression?  I.e. it looks like it would (no longer be supported/supportable) but it is also already broken in the master.   As is, it looks like it would "silently" executed the wrong thing (passing ```nullptr``` instead of the value of ```h1arr```.   Maybe we ought to fallback on the old code (when the variable is not found) and delegated to @Axel-Naumann or @vgvassilev to look into why the argument evaluation is no longer working properly. 

(Also it seems that passing a pointer value would not work either with the PR code).

Reply:
It works the way it is in this PR (with the `Add` context menu), when both parameters names are existing object (e.g. TH1F) in memory. As I said, I have a solution for the failing `TClingCallFunc::SetFunc()`, but it's not good enough. Another solution would be to completely remove this type of functionality from the GUI, even if I don't like the idea. But since it is broken from the very beginning of ROOT 6...
I can elaborate more next week if you want

Reply:
For now and for us to be able to proceed/merge this PR, can you add a fall back that does the old behavior is the address is not found?

Reply:
You mean this behavior?
```
Error in <TClingCallFunc::IFacePtr(kind)>: Attempt to get interface while invalid.
Error in <TClingCallFunc::Exec(address, interpVal)>: Called with no wrapper, not implemented!
```

Reply:
I was going to add code to print a meaningful error message instead (or maybe a message box, since it's a GUI functionality), that's why I put the `work in progress` and `do not merge` flags

Reply:
> You mean this behavior?

Yep :) or at least a way for Axel/Vassil to go ahead and debug in the inner guts of cling to improve the situation.
(and indeed at the very at least a nice error rather than silently doing the wrong thing ;) ).

Reply:
I know what is going wrong, I just don't know how to fix it properly (in cling/llvm/clang). And I prefer a meaningful message, since I can explain to @Axel-Naumann and @vgvassilev what's going on and why it doesn't work (I spent quite some time on it already). But let's wait until Axel is back, this PR is not urgent anyway (it's broken since 2014)